documentation/modules/ROOT/pages/connectors/mysql.adoc

```diff
@@ -373,90 +374,216 @@ For more information, see xref:mysql-schema-history-topic[schema history topic].
+. Produces a `read` event for each row that is captured from a table.
+   All `read` events contain the same binlog position, which is the position that was obtained in step 5.
+. Emits each `read` event to the Kafka topic for the source table.
+. Releases data table locks, if applicable.
 
 |8
-a|Commits the transaction.
+a|Commit the transaction.
 
 |9
-|Releases the table-level locks.
+|Release the table-level locks.
+Other database clients can now write to any previously locked tables.
 
 |10
-a|Records the completed snapshot in the connector offsets.
+a|Record the successful completion of the snapshot in the connector offsets.
 
 |===
 
+// ModuleID: {context}-description-of-why-initial-snapshots-capture-the-schema-history-for-all-tables
+// Title: Description of why initial snapshots capture the schema history for all tables
+// Type: concept
+[id="understanding-why-initial-snapshots-capture-the-schema-history-for-all-tables"]
+==== Understanding why initial snapshots capture the schema history for all tables
+
+When a connector runs an initial snapshot, it captures two types of information:
+
+Table data::
+Information about `INSERT`, `UPDATE`, and `DELETE` operations in tables that are designated by the connector's xref:{context}-property-snapshot-include-collection-list[`snapshot.include.collection.list`] property.
+Schema data::
+DDL statements that describe changes that occur to the structure of tables.
+
+By default, the snapshot captures schema data for every table that is present in the database, not only from tables that are designated for capture.
+For tables that are not designated for capture, the snapshot captures only the table structure.
+It does not capture any table data.
+The captured schema data is persisted to the both the internal schema history topic, and to the schema change topic, if one is configured.
+
+Users might be surprised to see that an intial snapshot captured schema information for non-captured tables, and might assume that an error occurred.
+But inital snapshots capture schema information for all tables by design.
+By persisting schema data for tables that are not part of the original capture set, the connector prepares itself to readily capture event data from these tables should that later become necessary.
+
+Because the database schema can change at any time, {prodname} must be able to identify the schema that is in effect for a table at the time that any event occurs.
+Then, if the connector later stops, when it resumes reading from the binlog, it can correlate event information with information from the schema history topic to ensure that it applies the correct schema to each event.
+
+If you exclude schema capture for non-captured tables, and you later want to capture data from such a table, an exception results due to the unknown schema.
+Additional manual steps are required to add the new table's schema to the schema history topic.
```

Comment:
I found some references in Zulip to how to add the schema for a table when the initial snapshot didn't capture it  (e.g., Delete the existing history topic, and then use schema_only_recovery to repopulate it) , but it isn't clear to me whether we want to include them in the documentation, and whether the steps might be consistent across connectors.

Reply:
These are consistent across all connectors sans PostgreSQL which doesn't have a schema history implementation; however, the caveat here remains that in order to recover the schema history, there cannot have been any schema changes made to the database between the time position of the transaction log offset and now.  In other words, when the connector re-reads the schemas using the JDBC metadata, the table structures must be identical to the expected table structures that all remaining transaction log entries are based on.  Make sense?